tooling/bin/assess-minibroker-charts.rb

```diff
@@ -0,0 +1,622 @@
+  FileUtils.mkdir_p(archive_saved)
+  FileUtils.cp(archive_patched, dst)
+end
+
+def regenerate_working_index
+  # Extract the chart blocks for all working charts from the master,
+  # patch the archive location to refer to the internal dev helm
+  # repository used by the brain tests, and save to a file.
+  working = {}
+  engines.each do |engine|
+    working[engine] = []
+    master_index[engine].each do |chart|
+      chartv = chart['version']
+      next unless @state && @state[engine] && @state[engine][chartv] && @state[engine][chartv]['works']
+      new = chart.dup
+      new['created'] = fixed_time
+      new['urls'] = "#{mbbt_repository}/#{engine}-#{chartv}.tgz"
+      working[engine] << new
+    end
+  end
+  File.write(working_saved, working.to_yaml)
+end
+
+# ......................................................................
+# Various (semi)constant values, mostly paths and the like
+
+def mbbt_repository
+  "https://minibroker-helm-charts.s3.amazonaws.com/kubernetes-charts"
+end
+
+def fixed_time
+  "2018-07-30T17:55:01.330815339Z"
+end
+
+def stable
+  "https://kubernetes-charts.storage.googleapis.com"
+end
+
+def helm_app
+  "chart-under-test"
+end
+
+def chart_in_app
+  "chart.tgz"
+end
+
+def statepath
+  @statepath = File.join(@workdir, 'results.yaml') unless @statepath
+  @statepath
+end
+
+def appdir
+  @appdir = File.join(@workdir, "charts") unless @appdir
+  @appdir
+end
+
+def working_saved
+  @wsaved = File.join(archive_saved, 'index.yaml') unless @wsaved
+  @wsaved
+end
+
+def archive_saved
+  @asaved = File.join(@workdir, 'ok') unless @asaved
+  @asaved
+end
+
+def archive_orig
+  @aunpatched = File.join(@workdir, 'archive-orig.tgz') unless @aunpatched
+  @aunpatched
+end
+
+def archive_patched
+  @apatched = File.join(@workdir, 'archive-patched.tgz') unless @apatched
+  @apatched
+end
+
+def manifest
+  @manifest = File.join(appdir, "manifest.yml") unless @manifest
+  @manifest
+end
+
+def chart_index
+  @chartindex = File.join(appdir, "index.yaml") unless @chartindex
+  @chartindex
+end
+
+def archive_app
+  @aapp = File.join(appdir, chart_in_app) unless @aapp
+  @aapp
+end
+
+def domain
+  @domain, _, _ = capture "kubectl get pods -o json --namespace \"#{@namespace}\" api-group-0 | jq -r '.spec.containers[0].env[] | select(.name == \"DOMAIN\").value'" unless @domain
+  @domain
+end
+
+def tester
+  @brain = File.join(@top, "make/tests") unless @brain
+  @brain
+end
+
+def appsrc
+  @appsrc = File.join(@top, "src/scf-release/src/acceptance-tests-brain/test-resources/node-env") unless @appsrc
+  @appsrc
+end
+
+def helm_repo
+  @helmrepo = "http://#{helm_app}.#{domain}" unless @helmrepo
+  @helmrepo
+end
+
+def target
+  @target = "https://api.#{domain}" unless @target
+  @target
+end
+
+def chart_ref
+  @chartref = "#{helm_repo}/#{chart_in_app}" unless @chartref
+  @chartref
+end
+
+# ......................................................................
+# Logging, and terminal cursor control
```

Comment:
Consider splitting bits into separate files (maybe spawning subprocess stuff can be one, and logging / terminal colouring can be another?). This file is rather large and disorganized :)

Reply:
will do

Reply:
At this point all the small stuff has been done and pushed to https://github.com/SUSE/scf-minibroker/pull/1